### DISCUSSION TITLE
Using Perplexity AI API with attached files

### DISCUSSION THREAD

**Discussion by asangare05** [perplexityai/api-discussion#76] [Q&A]
Hi,
I am trying to use the Perplexity AI API to send a request with attached files. My goal is to analyze the content of a PDF file via the API. However, when I attempt to attach a file through a URL or include the file in the body of the request, I encounter difficulties.

I tried sending the file's URL directly in the prompt, but the API doesn't seem to correctly process the content of the file. I'm unsure whether Perplexity AI directly supports attaching files in requests or via external URLs.

Is there a specific method for sending files to the API, or do I need to manually extract the content of the file and send it as text? Are there any limitations or additional configurations to consider when working with files and the API?

Thanks for any help or suggestions!
Abdoulaye

**Comment by shubhang98** [comment]
We don't have file uploads in the API today. You will have to send in the text of your PDF in the prompt.

**Reply by asangare05** [reply]
It's a shame that the api can't handle file uploads

**Comment by balach** [comment]
I have a similar problem. I actually converted my context text into a json object.. the api seems to disregard the stringified version of the object when passed into the system or user prompt.. 
use case: user gives an invoice => converted to json object => passed to api with the user's question: e.g. "what is the total invoice amount?" 
The json string contains "total amount":"$100" etc.. 
Would be super nice to have an additional field in request body called context in which to pass stringified version of json objects or just texts"
I get the right answer in perplexity gui if I paste the question + here's the object "{{stringified object}}" :/

**Comment by iqmeta** [comment]
Same. no pdf files - no need to use the api
